ledger-api/grpc-definitions/com/daml/ledger/api/v2/state_service.proto

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v2;
+
+import "com/daml/ledger/api/v1/event.proto";
+import "com/daml/ledger/api/v1/ledger_offset.proto";
+import "com/daml/ledger/api/v2/reassignment.proto";
+import "com/daml/ledger/api/v2/transaction_filter.proto";
+
+option java_outer_classname = "StateServiceOuterClass";
+option java_package = "com.daml.ledger.api.v2";
+option csharp_namespace = "Com.Daml.Ledger.Api.V2";
+
+// Allows clients to get state from the ledger.
+service StateService {
+
+  // Returns a stream of the snapshot of the active contracts and incomplete reassignments at a ledger offset.
+  // If there are no active contracts, the stream returns a single response message with the offset at which the snapshot has been taken.
+  // Clients SHOULD use the offset in the last GetActiveContractsResponse message to continue streaming transactions with the update service.
+  // Clients SHOULD NOT assume that the set of active contracts they receive reflects the state at the ledger end.
+  rpc GetActiveContracts (GetActiveContractsRequest) returns (stream GetActiveContractsResponse);
+
+  // Get the list of connected domains at the time of the query.
+  rpc GetConnectedDomains (GetConnectedDomainsRequest) returns (GetConnectedDomainsResponse);
+
+  // Get the current ledger end.
+  // Subscriptions started with the returned offset will serve events after this RPC was called.
+  rpc GetLedgerEnd (GetLedgerEndRequest) returns (GetLedgerEndResponse);
+}
+
+message GetActiveContractsRequest {
+
+  // Templates to include in the served snapshot, per party.
+  // Required
+  TransactionFilter filter = 1;
+
+  // If enabled, values served over the API will contain more information than strictly necessary to interpret the data.
+  // In particular, setting the verbose flag to true triggers the ledger to include labels for record fields.
+  // Optional
+  bool verbose = 2;
```

Comment:
@andreaslochbihler-da @nmarton-da : consider removing the verbose mode from the API level, as
1. verbose mode requires extra thought with the upcoming upgrading proposal
2. customers that need field names and the like can use the JSON API port exposed by Canton
3. adding the feature back upon customer request is easy to do, as we'll still have it in our codebase to drive the v1 API

Reply:
Sounds great to me, quite some complexity would be removed in LAPI server. On the other hand it feels like this needs more consideration. Would it be okay to add it to the list to consider for 3.0 and keep it for now?

Reply:
Yes, definitely. All my review comments are only for your consideration. Your call how to deal with them best.